extensions/2.0/Khronos/KHR_materials_specular/README.md

```diff
@@ -0,0 +1,247 @@
+# KHR\_materials\_specular
+
+## Contributors
+
+* Tobias Haeussler, Dassault Systemes [@proog128](https://github.com/proog128)
+* Bastian Sdorra, Dassault Systemes [@bsdorra](https://github.com/bsdorra)
+* Ed Mackey, AGI [@emackey](https://twitter.com/emackey)
+* Richard Sahlin, IKEA [@rsahlin](https://github.com/rsahlin)
+* Ben Houston, ThreeKit [@bhouston](https://github.com/bhouston)
+* Emmett Lalish, Google [@elalish](https://github.com/elalish)
+* Don McCurdy, Google [@donrmccurdy](https://twitter.com/donrmccurdy)
+* Norbert Nopper, UX3D [@UX3DGpuSoftware](https://twitter.com/UX3DGpuSoftware)
+* Sebastien Vandenberghe, Microsoft [@sebavanjs](https://twitter.com/sebavanjs)
+
+Copyright (C) 2018-2020 The Khronos Group Inc. All Rights Reserved. glTF is a trademark of The Khronos Group Inc.
+See [Appendix](#appendix-full-khronos-copyright-statement) for full Khronos Copyright Statement.
+
+## Status
+
+Experimental
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Exclusions
+
+* This extension must not be used on a material that also uses `KHR_materials_pbrSpecularGlossiness`.
+* This extension must not be used on a material that also uses `KHR_materials_unlit`.
+
+## Overview
+
+This extension adds two parameters to the metallic-roughness material: `specular` and `specularColor`.
+
+`specular` allows users to configure the strength of the specular reflection in the dielectric BRDF. A value of zero disables the specular reflection, resulting in a pure diffuse material. The metal BRDF is not affected by the parameter.
+
+`specularColor` changes the F0 color of the specular reflection in the dielectric BRDF, allowing artists to use effects known from the specular-glossiness material (`KHR_materials_pbrSpecularGlossiness`) in the metallic-roughness material.
+
+## Extending Materials
+
+The strength of the specular reflection is defined by adding the `KHR_materials_specular` extension to any glTF material.
+
+```json
+{
+    "materials": [
+        {
+            "extensions": {
+                "KHR_materials_specular": {
+                    "specularFactor": 1.0,
+                    "specularColorFactor": [1.0, 1.0, 1.0],
+                }
+            }
+        }
+    ]
+}
+```
+
+Factor and texture are combined by multiplication to describe a single value.
+
+| |Type|Description|Required|
+|-|----|-----------|--------|
+| **specularFactor** | `number` | The strength of the specular reflection. | No, default: `1.0`|
+| **specularColorFactor** | `number[3]` | The F0 color of the specular reflection (RGB). | No, default: `[1.0, 1.0, 1.0]`|
+| **specularTexture** | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo) | A 4-channel texture that defines the F0 color of the specular reflection (RGB channels, encoded in sRGB) and the specular factor (A). Will be multiplied by specularFactor and specularColorFactor. | No |
```

Comment:
I think the texture should be split into two parts:
specularColor and specularFactor (or strength/specularStrength as suggested above) to be consisten with what we previously have defined.

Reply:
My preference would be to more explicitly define the behavior of a 3-channel image when supplied here.  Namely, the alpha channel is presumed to be all 1.0, meaning it's a no-op multiplication with the corresponding factor that can be optimized out.

Reply:
As discussed in the PBR call yesterday I have checked with our material experts for a usecase where two texture infos are needed:
A dielectric material, eg rubber, that has a normal texture (RGB) and use the specularFactorTexture (A) to change the specular/diffuse contribution over the mesh (without changing the color).

I don't see how that is possible with the current specification.

Reply:
Thanks for the specific example, @rsahlin. Correct me if I'm wrong, but wouldn't this material be well represented by transforming the specularFactor into the metallic channel? Specular only differs from metallic when the specular color differs from the base color. Otherwise it is basically just metallic (which is why we're pushing the spec-gloss workflow toward metal-rough).

Reply:
Well, there is a difference in how the refracted light behaves in metals as opposed to dielectrics, ie all refracted light is absorbed in the metal.

Shouldn't this be taken into account?

Reply:
Okay, so it sounds like this is a different example, specifically a transparent metal, correct? Can you describe the type of real material you're hoping to represent with this? I think we had talked about how this particular combination didn't tend to make a lot of sense physically. Also, if you really want to control specular strength, could it be IOR that you're actually after?

Reply:
I'm just thinking that for a dielectric the scattering is different - so it may make sense to be able to control the factor.
I have to delve deeper into this with our technical artists.

Ok, then consider the opposite:
The factor does not change along the mesh, only the RGB - this gives the option of having 1 channel from another material stored here.
For instance the occlusion channel.

Reply:
@rsahlin With "change the specular/diffuse contribution over the mesh" you mean that you want to decrease specular contribution and increase diffuse contribution, right? And if you would use the metallic channel, as proposed by @elalish, you would get the opposite: more specular than diffuse, which isn't what you want?

Reply:
@proog128 Yes, for instance a teacup made of porcelain.
The color of the object and the color of the specular reflection is the same but the diffuse/specularity factor may vary over the surface.

I've had a chat with a professor of Computer Graphics at the university in Linköping (who understands glTF) and in his opinion it makes sense to split these up.

Reply:
Can we see this teacup model? Presumably this is currently modeled with the spec-gloss extension? I think it would be very helpful to have a few example models to show how best to convert them into metal-rough with this specular extension. I still think we can do a good job with the metallic and roughness factors, but we can only be sure if we try and view the results side-by-side.

Reply:
@elalish This was a discussion I had with the (theoretical) material experts at Ikea.
To find a model that would use this I have to ask production.

I would say that we will not get the same result with the MR properties due to the absorption properties of metals.
Just look at the metallicroughness spheres and I think you will see what I mean.
Compare the bottom row (dielectric) with the top row (metal) - as we go forward, with for instance the SSS extension, the dielectrics is what will contribute to the diffuse part.
The change I am proposing will let us control this property, without having to modulate the RGB (and thus potentially 'saving' 3 channels when specular color does not change) - while not changing the roughness (ie keeping the 'sharpness' of the specular highlight)

![image](https://user-images.githubusercontent.com/3063192/94892744-b7752000-0485-11eb-837d-c9d1d95e8961.png)

Reply:
After speaking with a material expert from the production:
Basically all our wood materials use a speculargloss material with a monochrome (grey) RGB texture and the 'intensity' channel specified per pixel (which relates to the specular factor in this extension).
With my propsed change all these materials will be able to drop the RGB channels.

Reply:
@rsahlin Perfect, can you send some of these wood models? I want to make a tool for converting spec-gloss models to metal-rough and I'd like to see how well the conversion can be done without needing the specular extension. I know it won't work all the time, but I think this wood example would be a great way to figure out how far we can go. 

Also, what viewer did you use for that last screenshot? It's not even faintly spec-compliant, as it does not seem to read the roughness value at all! Compare to the others [here](https://modelviewer.dev/fidelity/).

Reply:
@elalish I'm sorry but Ikea is super strict about sending models to outside parties, it's currently not possible.

Yes, the image does not look very good does it?
It's taken from the screenshot folder in the sample model repo :-)